### PR TITLE
Implement `cudf::copy_if_else` for `decimal32` and `decimal64`

### PR DESCRIPTION
**Pull request by codereport**
This PR resolves a part of https://github.com/rapidsai/cudf/issues/3556.

**To Do List:**
* [x] Implementation
* [x] Basic unit tests
* [x] Comprehensive unit tests

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Review by harrism — APPROVED** [review]
Looks good.

File: cpp/include/cudf/detail/copy_if_else.cuh [hunk 1]
```diff
@@ -163,6 +163,7 @@ std::unique_ptr<column> copy_if_else(
   LeftIter lhs_end,
   RightIter rhs,
   FilterFn filter,
+  cudf::data_type output_type,
```
Comment (rgsl888prabhu) [inline cpp/include/cudf/detail/copy_if_else.cuh:166]: Missing doc update

**Review by rgsl888prabhu — APPROVED** [review]
LGTM, just missing doc update and header  change from 2019 -> 2019-2020

**Comment by codereport** [comment]
rerun tests

**Pull request merged**